src/meta/src/hummock/manager/gc.rs

```diff
@@ -121,14 +121,14 @@ impl HummockManager {
             let versioning_guard = read_lock!(self, versioning).await;
             let versioning: &Versioning = &versioning_guard;
 
-            // object ids in current version
-            let mut tracked_object_ids = versioning.current_version.get_object_ids();
-            // add object ids removed between checkpoint version and current version
+            // object ids in checkpoint version
+            let mut tracked_object_ids = versioning.checkpoint.version.get_object_ids();
+            // add object ids added between checkpoint version and current version
             for (_, delta) in versioning.hummock_version_deltas.range((
                 Excluded(versioning.checkpoint.version.id),
                 Included(versioning.current_version.id),
             )) {
-                tracked_object_ids.extend(delta.gc_object_ids.iter().cloned());
+                tracked_object_ids.extend(delta.newly_added_object_ids());
```

Comment:
just a question: why we need to track gc_object_ids before? This seems counterintuitive

Reply:
> Currently we use the gc_object_ids in the hummock version delta to record the sst object ids being removed in the new version. We have some pinned versions older than the current version, and we need to track the object in these previous version so that we won't delete these objects by accidence, and with gc_object_ids we can easily achieve this.

I guess previously the `gc_object_ids` is persisted so that we don't have to regenerate the `gc_object_ids` of each version delta.